src/main/java/frc/robot/subsystems/arm/ArmIOReal.java

```diff
@@ -0,0 +1,324 @@
+    private boolean changedToDefaultCommand = false;
+
+    private double ySetpoint = 0;
+
+    private ControlRequest motorControlRequest = null;
+
+
+
+
+    ArmIOReal() {
+
+        this.shoulderEncoder = new DutyCycleEncoder(Ports.ArmPorts.SHOULDER_ENCODER);
+        this.elbowEncoder = new DutyCycleEncoder(Ports.ArmPorts.ELBOW_ENCODER);
+
+        this.shoulderMainMotor = new TalonFX(Ports.ArmPorts.SHOULDER_MAIN_MOTOR);
+
+        this.shoulderAuxMotor = new TalonFX(Ports.ArmPorts.SHOULDER_AUX_MOTOR);
+
+        this.elbowMainMotor = new TalonFX(Ports.ArmPorts.ELBOW_MAIN_MOTOR);
+
+        this.elbowAuxMotor = new TalonFX(Ports.ArmPorts.ELBOW_AUX_MOTOR);
+
+
+
+        shoulderAuxMotor.setControl(new StrictFollower(shoulderMainMotor.getDeviceID()));
+        elbowAuxMotor.setControl(new StrictFollower(elbowMainMotor.getDeviceID()));
+
+
+        shoulderMainMotorConfig.Voltage.PeakForwardVoltage = ArmConstants.VOLT_COMP_SATURATION;
+        shoulderMainMotorConfig.Voltage.PeakReverseVoltage = -ArmConstants.VOLT_COMP_SATURATION;
+        shoulderMainMotorConfig.Slot0 = SHOULDER_PID;
+        elbowMainMotorConfig.CurrentLimits.SupplyCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.CurrentLimits.StatorCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        shoulderMainMotorConfig.Feedback.SensorToMechanismRatio = SENSOR_TO_MECHANISM_RATIO;
+        shoulderMainMotorConfig.Feedback.RotorToSensorRatio = ROTOR_TO_SENSOR_RATIO;
+        shoulderMainMotor.getConfigurator().apply(shoulderMainMotorConfig);
+
+        shoulderMainMotor.setNeutralMode(NeutralModeValue.Brake);
+        shoulderMainMotor.setInverted(MAIN_CLOCKWISE);
+
+        elbowMainMotorConfig.Voltage.PeakForwardVoltage = ArmConstants.VOLT_COMP_SATURATION;
+        elbowMainMotorConfig.Voltage.PeakReverseVoltage = -ArmConstants.VOLT_COMP_SATURATION;
+        elbowMainMotorConfig.Slot1 = ELBOW_PID;
+        elbowMainMotorConfig.CurrentLimits.SupplyCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.CurrentLimits.StatorCurrentLimit = ArmConstants.CURRENT_LIMIT;
+        elbowMainMotorConfig.Feedback.SensorToMechanismRatio = SENSOR_TO_MECHANISM_RATIO;
+        elbowMainMotorConfig.Feedback.RotorToSensorRatio = ROTOR_TO_SENSOR_RATIO;
+        elbowMainMotor.getConfigurator().apply(elbowMainMotorConfig);
+
+        elbowMainMotor.setNeutralMode(NeutralModeValue.Brake);
+        elbowMainMotor.setInverted(MAIN_CLOCKWISE);
+
+    }
+
+    /**
+     * Get the instance of the arm subsystem.
+     *
+     * @return Arm instance
+     */
+    public static ArmIOReal getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ArmIOReal();
+        }
+        return INSTANCE;
+    }
+
+
+    /**
+     * Sets the power of the shoulder motors.
+     *
+     * @param power desired power. [-1,1]
+     */
+    public void setShoulderJointPower(double power) {
+        shoulderMainMotor.setControl(new DutyCycleOut(power));
```

Comment:
I think that I confused everyone. Every kind of control request for every motor should be a member in the class. When passing the request to the motor, the `with` functions should be used. Like this:
```suggestion
        shoulderMainMotor.setControl(dutyCycleShoulderRequest.withOutput(power).withEnableFOC(true));
```

Reply:
True, just that `withEnableFOC(true)` can be called once on instance creation and you don't need to call this every time.